src/main/java/org/apache/solr/benchmarks/solrcloud/SolrCloud.java

```diff
@@ -180,6 +184,38 @@ public void init() throws Exception {
     	for (String host: getSolrNodesFromVagrant()) {
     		nodes.add(new GenericSolrNode(host, null)); // TODO fix username for vagrant
     	}
+    } else if ("external".equalsIgnoreCase(cluster.provisioningMethod)) {
+        log.info("ZK node: " + cluster.externalSolrConfig.zkHost);
+        String[] tokens = cluster.externalSolrConfig.zkHost.split(":");
+        zookeeper = new GenericZookeeper(tokens[0], Integer.parseInt(tokens[1]), cluster.externalSolrConfig.zkAdminPort, cluster.externalSolrConfig.zkChroot);
+
+        try (CloudSolrClient client = new CloudSolrClient.Builder().withZkHost(cluster.externalSolrConfig.zkHost).withZkChroot(cluster.externalSolrConfig.zkChroot).build()) {
+          Set<String> liveNodes = client.getZkStateReader().getClusterState().getLiveNodes();
+          for (String liveNode: liveNodes) {
+            nodes.add(new ExternalSolrNode(
+                    liveNode.split("_solr")[0].split(":")[0],
+                    Integer.valueOf(liveNode.split("_solr")[0].split(":")[1]),
+                    cluster.externalSolrConfig.sshUserName,
+                    cluster.externalSolrConfig.restartScript));
+          }
+          try {
+            List<String> queryNodeList = client.getZkStateReader().getZkClient().getChildren("/live_query_nodes", null, true);
```

Comment:
Probably we need now coordinator role here now.  CCing @chatman

Reply:
Yes, that is a good point, Hitesh. I think we should support both. If "/live_query_nodes" exists, use it, otherwise look for coordinator nodes. I'll raise a separate PR for this.